data/maps/DesertUnderpass/scripts.inc

```diff
@@ -6,29 +6,31 @@ DesertUnderpass_OnTransition: @ 823AF3D
 	setflag FLAG_LANDMARK_DESERT_UNDERPASS
 	end
 
-DesertUnderpass_EventScript_23AF41:: @ 823AF41
+DesertUnderpass_EventScript_Fossil:: @ 823AF41
 	lock
 	faceplayer
-	goto_if_set FLAG_CHOSE_CLAW_FOSSIL, DesertUnderpass_EventScript_23AF57
-	goto_if_set FLAG_CHOSE_ROOT_FOSSIL, DesertUnderpass_EventScript_23AF68
+	goto_if_set FLAG_CHOSE_ROOT_FOSSIL, DesertUnderpass_EventScript_GiveClawFossil
```

Comment:
`goto_if_set FLAG_CHOSE_ROOT_FOSSIL, DesertUnderpass_EventScript_GiveClawFossil`?

Reply:
Correct. The fossil in the Desert Underpass is whatever fossil wasn't chosen in Mirage Tower. The `FLAG_CHOSE_X_FOSSIL` flag was incorrect prior to this